lib/material_datepicker/preset.dart

```diff
@@ -0,0 +1,173 @@
+          new WeekRange.weeksAgo(clock, 0, startWeekday: startDay),
+          shortTitle: _thisWeekShortTitle(startDayName),
+          alternatives: alternatives);
+      alternatives.add(preset);
+      if (startDay == startWeekday) result = preset;
+    }
+    return result;
+  }
+
+  /// Create a [DatepickerPreset] for selecting "Last week".
+  ///
+  /// [startWeekday] is the 1-based first day of the week: e.g.
+  /// [DateTime.MONDAY] (which equals 1) for Monday, [DateTime.TUESDAY] for
+  /// Tuesday, etc. It defaults to the first day of the week specified in the
+  /// current locale.
+  ///
+  /// [validStartWeekdays] is a list of all starting weekdays which will be
+  /// added as [alternatives].
+  factory DatepickerPreset.lastWeek(Clock clock,
+      {int startWeekday, List<int> validStartWeekdays}) {
+    startWeekday ??= _defaultStartWeekday;
+    validStartWeekdays =
+        _initValidStartWeekdays(startWeekday, validStartWeekdays);
+
+    DatepickerPreset result;
+    List<DatepickerPreset> alternatives = [];
+    for (var startDay in validStartWeekdays) {
+      var endDay = 1 + ((startDay - 1) + 6).remainder(7);
+      var startDayName = _weekdayName(startDay);
+      var endDayName = _weekdayName(endDay);
+      var preset = new DatepickerPreset(
+          _lastWeekTitle(startDayName, endDayName),
+          new WeekRange.weeksAgo(clock, 1, startWeekday: startDay),
+          shortTitle: _lastWeekShortTitle(startDayName, endDayName),
+          alternatives: alternatives);
+      alternatives.add(preset);
+      if (startDay == startWeekday) result = preset;
+    }
+    return result;
+  }
+
+  /// Create a [DatepickerPreset].
+  DatepickerPreset(this.title, this.range,
+      {this.shortTitle, this.alternatives});
+
+  static final _weekdayNames =
+      new DateFormat().dateSymbols.STANDALONESHORTWEEKDAYS;
+
+  // Where 1 = Mon, 7 = Sun.
+  static int get _defaultStartWeekday =>
+      1 + new DateFormat().dateSymbols.FIRSTDAYOFWEEK;
+
+  static const _defaultValidStartWeekdays = const [
+    DateTime.SUNDAY,
+    DateTime.MONDAY
+  ];
+
+  static List<int> _initValidStartWeekdays(
+      int startWeekday, List<int> validStartWeekdays) {
+    validStartWeekdays ??= new List<int>.from(_defaultValidStartWeekdays);
+    if (!validStartWeekdays.contains(startWeekday)) {
+      validStartWeekdays.insert(0, startWeekday);
+    }
+    return validStartWeekdays;
+  }
+
+  // [weekday] is [1 = Mon, 7 = Sun] and [_weekdayNames] is [Sun .. Sat].
+  static String _weekdayName(int weekday) =>
+      _weekdayNames[weekday.remainder(7)];
+
+  static String _thisWeekTitle(String startDayName) =>
+      Intl.message('This week ($startDayName – Today)',
+          name: 'DatepickerPreset__thisWeekTitle',
+          desc: 'Label for a date range corresponding to "this week" starting '
+              'on a particular day of the week and ending today.'
+              ' [REL_NOTE: zjd/2017-12-31]',
```

Comment:
Are the REL_NOTE's relevant? (There are a few in this file.)

Reply:
I'll check with author and remove in the future if needed.